homeassistant/components/plugwise/number.py

```diff
@@ -88,43 +90,34 @@ def __init__(
         coordinator: PlugwiseDataUpdateCoordinator,
         device_id: str,
         description: PlugwiseNumberEntityDescription,
+        actuator: ActuatorData,
```

Comment:
I'm not following the logic behind passing this data?

The description is available already, so the class instance can already instance it without passing and storing this again?

../Frenck

Reply:
We could call it using `self.device[self.entity_description.key]` instead of passing through the additional but that would throw the Typing Literal requireds back in? (IIRC)

Reply:
@frenck this change was suggested by epenet:
![image](https://github.com/home-assistant/core/assets/11290930/d953b601-24b2-407f-a0ef-418554a3f98a)

Reply:
It is compulsory to run `actuator_fn` in the `async_setup_entry` loop.
We then have three options:
- don't pass the instance, don't create the instance in the entity constructor,  but run the `actuator_fn` again in each property
- don't pass the instance, but run the `actuator_fn` again in the entity constructor
- pass the instance to the entity constructor

I felt the third option was best.
@frenck are you suggesting that you prefer the second option?

Reply:
No I meant 1.

Reply:
@frenk Option 1 has been implemented, please review.

Reply:
I've used a double-if construct:
```
            if device.get(description.key) is not None:
                if device[description.key].get("setpoint") is not None:
```
with the idea to be able to remove the 2nd line when we upgrade to the latest backend in a next PR.

Reply:
@epenet if you have time, please review as well, thanks!

Reply:
@bouwew you didn't implement option 1, you implemented option 2
Frenck explicitely requested option 1

Reply:
@epenet I did implement option 1, I've removed the actuator_fn lambda. I did reuse actuator in `self.actuator = self.device[description.key]`.

Reply:
On second thought, I've implemented yet another option, without `actuator_fn`. I realized I could make the code more generic, as was Frenck's intention (I think).